tests/wpunit/Admin/AdminTest.php

```diff
@@ -0,0 +1,135 @@
+use WPBDP;
+use Codeception\Util\Debug;
+
+/**
+ * Tests for the Admin.
+ *
+ * @since x.x
+ */
+class AdminTest extends WPUnitTestCase {
+
+	/**
+	 * WpunitTester instance.
+	 *
+	 * @var \WpunitTester
+	 */
+	protected $tester;
+
+	/**
+	 * WPBDP_Admin instance.
+	 *
+	 * @var \WPBDP_Admin
+	 */
+	private static $admin;
+
+	/**
+	 * Admin user ID.
+	 *
+	 * @var int
+	 */
+	protected static $admin_user_id;
+
+	/**
+	 * Editor user ID.
+	 *
+	 * @var int
+	 */
+	protected static $editor_user_id;
+
+	/**
+	 * WPBDP menu ID.
+	 *
+	 * @var string
+	 */
+	protected static $menu_id;
+
+	public static function wpSetUpBeforeClass( $factory ) {
+		static::$admin_user_id = $factory->user->create( [ 'role' => 'administrator' ] );
+		static::$editor_user_id = $factory->user->create( [ 'role' => 'editor' ] );
+
+		if ( ! class_exists( 'WPBDP_Admin' ) ) {
+			require_once WPBDP_INC . 'admin/class-admin.php';
+		}
+
+		static::$admin = new WPBDP_Admin();
+		static::$menu_id = static::$admin->get_menu_id();
+	}
+
+	public function testMenuExistsAsAdmin() {
+		$this->tester->wantToTest( 'Admin Menu Exists As Admin' );
+		$this->wpbdp_admin_menu_exists( static::$admin_user_id );
+	}
+
+	public function testMenuExistsAsEditor() {
+		$this->tester->wantToTest( 'Admin Menu Exists For Editor User' );
+		$this->wpbdp_admin_menu_exists( static::$editor_user_id );
+	}
+
+	public function testListingPostTypeDoesNotExistAtTopLevelAsAdmin() {
+		$this->tester->wantToTest( 'Listing Post Type Does Not Exist At Top Level As Admin' );
+		$this->wpbdp_listing_post_type_at_top_level_exists( static::$admin_user_id );
+	}
+
+	public function testCombineListingPostTypeAndWPBDP_Admin() {
+		global $submenu;
+		$original_submenu = $submenu;
+
+		wp_set_current_user( static::$editor_user_id );
+		static::$admin->admin_menu();
+		static::$admin->hide_menu();
+
+		$submenu = $original_submenu;
+	}
+
+	public function testSubmenusExistAsAdmin() {
+		global $submenu;
+		$original_submenu = $submenu;
+
+		wp_set_current_user( static::$admin_user_id );
+		set_current_screen( 'index.php' );
+		$this->assertTrue( is_admin() );
+
+		$themes = new WPBDP_Themes();
+
+		static::$admin->admin_menu();
+		static::$admin->hide_menu();
+
+		$this->assertEquals( 'wpbdp_settings', $submenu[ static::$menu_id ][2][2] );
+		$this->assertEquals( 'wpbdp_admin_payments', $submenu[ static::$menu_id ][5][2] );
+		$this->assertEquals( 'wpbdp-addons', $submenu[ static::$menu_id ][8][2] );
+		$this->assertEquals( 'wpbdp-themes', $submenu[ static::$menu_id ][9][2] );
+
+		$submenu = $original_submenu;
+	}
+
+	protected function wpbdp_admin_menu_exists( $user_id ) {
+		global $submenu;
+		$original_submenu = $submenu;
+
+		wp_set_current_user( $user_id );
+		static::$admin->admin_menu();
+		static::$admin->hide_menu();
+
+		$this->assertArrayHasKey( static::$menu_id, $submenu );
+
+		$submenu = $original_submenu;
+	}
+
+	protected function wpbdp_listing_post_type_at_top_level_exists( $user_id ) {
```

Comment:
@stephywells I couldn't find a way to check Listing Post Type as a top-level menu! Do you have any suggestions?

Reply:
Is this included in $menu? What does the menu include with var_dump($menu)?

Reply:
It returns this array:

```
Array
  (
      [20] => Array
          (
              [0] => Directory
              [1] => edit_posts
              [2] => wpbdp_admin
              [3] => Business Directory Admin
              [4] => menu-top toplevel_page_wpbdp_admin
              [5] => toplevel_page_wpbdp_admin
              [6] => ...
          )
  
      [20.06409] => Array
          (
              [0] => Directory
              [1] => edit_posts
              [2] => wpbdp_admin
              [3] => Business Directory Admin
              [4] => menu-top toplevel_page_wpbdp_admin
              [5] => toplevel_page_wpbdp_admin
              [6] => ...
          )
  )
```

Reply:
Is this the output after it runs through the menu combine function?

Reply:
Yes, I've tried possible conditions.

Reply:
It looks like you would loop through and check if [2] === 'wpbdp_admin'.

I'm seeing this if the extra menu isn't removed:
```
[5]=>
--
  | array(7) {
  | [0]=>
  | string(9) "Directory"
  | [1]=>
  | string(10) "edit_posts"
  | [2]=>
  | string(32) "edit.php?post_type=wpbdp_listing"
  | [3]=>
  | string(0) ""
  | [4]=>
  | string(46) "menu-top menu-icon-wpbdp_listing menu-top-last"
  | [5]=>
  | string(24) "menu-posts-wpbdp_listing"
  | [6]=> ...
  | }
```

This test needs to make sure 'edit.php?post_type=wpbdp_listing' is not included, and 'wpbdp_admin' is included.

Reply:
Where did you get this output?